### DISCUSSION TITLE
Show spinner while NuxtPage loading?

### DISCUSSION THREAD

**Discussion by pixelspark** [nuxt/nuxt#18003] [Questions] [answered]
I am using `<NuxtPage>` to show pages, inside of which I use `$router.replace(...)` to move between pages (should be similar to what `<NuxtLink>` is doing):

````vue
<Button @click="$router.replace('/' + x + '/' + y)" />
````

(Yes, I know I can probably improve this using parameters and named routes).

This works fine, but feels a bit 'laggy': when clicking the button, the page starts loading in the background, only to be swapped in half a second later. This is especially confusing when moving to the same route with different parameters (page looks like it is showing stale info). During the loading time I would therefore like to hide/blur the page and show a spinner. Is there a way to configure the content `<NuxtPage>` shows while it is switching pages?

**Comment by pixelspark** [comment, marked as answer]
Figured it out:

````ts
const nuxtApp = useNuxtApp();
const loading = ref(false);
nuxtApp.hook("page:start", () => {
	loading.value = true;
});

nuxtApp.hook("page:finish", () => {
	loading.value = false;
});
````

Then use with e.g. `:class="{loading}` or `v-if="loading"` on a spinner, etc.

It appears NuxtPage also wraps the page in a Vue `<transition>` tag which will set some classes, but these do not seem to work reliably when moving between pages. 

Documentation could use some updating!

**Reply by jarraga** [reply]
Hi pixelspark, thanks for your answer, that works for me! but where can I find that information? I can't find anything in nuxt 3 docs...

**Reply by pixelspark** [reply]
Yeah I think I ultimately got it from Nuxt v2 docs and tried if the same mechanism worked on v3 (combined with some browsing through the source of v3 to figure out what hooks/events are supported..)
